lighthouse-core/index.js

```diff
@@ -70,6 +70,10 @@ lighthouse.getAuditList = Runner.getAuditList;
 lighthouse.traceCategories = require('./gather/driver.js').traceCategories;
 lighthouse.Audit = require('./audits/audit.js');
 lighthouse.Gatherer = require('./gather/gatherers/gatherer.js');
+
+// Explicit type reference (hidden by makeComputedArtifact) for d.ts export.
+// TODO(esmodules): should be a workaround for module.export and can be removed when in esm.
+/** @type {typeof import('./computed/network-records.js')} */
```

Comment:
kind of a dumb workaround :) The need for it is _kind of_ covered by https://github.com/microsoft/TypeScript/issues/37832#issuecomment-611063969 and https://github.com/microsoft/TypeScript/issues/41672, but it's obviously not fixed and it's not a typedef causing the issue.

tl;dr: typescript creates an (ESM-ish) `export` object for the d.ts file, but needs a type for `NetworkRecords`. It then gets confused by the type because `NetworkRecords` is a computed artifact, which subsumes the original `NetworkRecords` type, but the original `NetworkRecords` is still needed because the computed artifact return type is `NetworkRecords & {request(artifacts: DevTools, cache: LH.Artifacts.ComputedContext): Promise<whatever>}`, hence the `"requires using private name 'NetworkRecords'"` complaint.

"Kind of dumb" because the annotation is exactly the expected type for the `require()`, but just need to explicitly type it to unconfuse the compiler and it should go aways with the conversion to es modules anyways, so 🤷